dev/tasks/main.go

```diff
@@ -0,0 +1,285 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+//  go run . --remote upstream  --branch upstream/master --version-file version/VERSION  --source ${REPO} -v=2 --yes=true
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+
+	// Support GCP auth
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/klog/v2"
+)
+
+func main() {
+	ctx := context.Background()
+	err := run(ctx)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		os.Exit(1)
+	}
+}
+
+func run(ctx context.Context) error {
+	log := klog.FromContext(ctx)
+
+	remote := ""
+	flag.StringVar(&remote, "remote", remote, "name of remote")
+	sourceCheckout := ""
+	flag.StringVar(&sourceCheckout, "source", sourceCheckout, "location of source code checkout")
+	branch := ""
+	flag.StringVar(&branch, "branch", branch, "branch for creation of tags")
+	versionFile := ""
+	flag.StringVar(&versionFile, "version-file", versionFile, "path to version marker file")
+	pushOptions := ""
+	flag.StringVar(&pushOptions, "push-options", pushOptions, "when pushing tags to git, include these options")
+	yes := false
+	flag.BoolVar(&yes, "yes", yes, "actually do it")
+	addVPrefix := false
+	flag.BoolVar(&addVPrefix, "add-v-prefix", addVPrefix, "prefix tag with 'v'")
+	klog.InitFlags(nil)
+	flag.Parse()
+
+	if branch == "" {
+		return fmt.Errorf("must specify --branch")
+	}
+
+	if remote == "" {
+		return fmt.Errorf("must specify --remote")
+	}
+
+	if sourceCheckout == "" {
+		return fmt.Errorf("must specify --source")
+	}
+
+	if versionFile == "" {
+		return fmt.Errorf("must specify --version-file")
+	}
+
+	homeDir, err := os.UserHomeDir()
```

Comment:
Why are we getting the user home directory? It seems like we should be using the current directory (https://pkg.go.dev/os#Getwd). We can ask people to run the script within the current directory. I don't think we should be making assumptions about where people checked out code.